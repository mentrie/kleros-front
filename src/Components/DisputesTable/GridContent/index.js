import React, { Component } from 'react'
import _ from 'lodash'
import { connect } from 'react-redux'
import { withRouter, Link } from 'react-router-dom'
import 'babel-polyfill'
import { getDisputes } from '../../../redux/disputes/action-creators'
import { truncateText } from '../../../helpers/truncateText'
import { STATUS_TO_STATE } from '../../../constants'
import './GridContent.css'

export class GridContent extends Component {
  componentWillMount () {
    this.props.getDataDisputes()
  }

  render () {
    const { hasErrored, isFetching, disputes, baseLink, filterFunction } = this.props
    if (hasErrored) {
      return <p>Sorry! There was an error loading the disputes</p>
    }

    if (isFetching) {
      return (
        <div className='GridContent-container'>
          <div className='items loader'>
            <div className='linear-background-100' />
            <div className='linear-background-90' />
            <div className='linear-background-90' />
          </div>
        </div>
      )
    }

    const filteredDisputes = filterFunction(disputes)

    return (
      <div className='GridContent-container'>
        <div className='items'>
          {
            _.isEmpty(disputes) &&
            <div className='items-row'>
              <div className='item item-no-disputes'>
                You have no disputes.
              </div>
            </div>
          }
          {
            filteredDisputes.filter(dispute => {
              return dispute.arbitratorAddress === process.env.REACT_APP_ARBITRATOR_ADDRESS
            }).map(dispute =>
              (
                <Link key={dispute.arbitrableContractAddress} to={`${baseLink}/${dispute.arbitrableContractAddress}`}>
                  <div className='items-row'>
                    <div className='item item-project'>
                      <div className='item-title'>{ truncateText(dispute.description ? dispute.description : 'unavailable', 35) }</div>
                      <div className='item-category'>{ dispute.category }</div>
                    </div>
                    <div className='item item-deadline'>{ dispute.deadline }</div>
                    <div className='item item-case_id'>{ truncateText(dispute.hash, 10) }</div>
                    <div className='item item-status'>{ STATUS_TO_STATE[dispute.arbitrableContractStatus] }</div>
                  </div>
                </Link>
              )
            )
          }
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    disputes: state.disputes.disputes,
    hasErrored: state.disputes.failureDisputes,
    isFetching: state.disputes.requestDisputes
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getDataDisputes: () => dispatch(getDisputes())
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(GridContent))
