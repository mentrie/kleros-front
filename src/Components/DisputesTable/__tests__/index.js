import React from 'react'
import DisputesTable from '../index'

describe('DisputesTable Exists', () => {
  const component = shallow(<DisputesTable />)
  test('should render component properly', () => {
    expect(component).toMatchSnapshot()
  })
})
