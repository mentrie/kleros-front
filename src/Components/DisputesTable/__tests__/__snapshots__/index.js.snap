// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisputesTable Exists should render component properly 1`] = `
<div
  className="Grid-container undefined"
>
  <div
    className="grid"
  >
    <div
      className="header"
    >
      <div
        className="item header-project-item"
      >
        Project
      </div>
      <div
        className="item header-deadline-item"
      >
        Deadline
      </div>
      <div
        className="item header-case_id-item"
      >
        Case ID
      </div>
      <div
        className="item header-status-item"
      >
        Status
      </div>
    </div>
    <withRouter(Connect(GridContent))
      filterFunction={[Function]}
    />
  </div>
</div>
`;
